src/Action/LoadStream.php

```diff
@@ -180,14 +170,10 @@ private function returnDescription(ServerRequestInterface $request, string $stre
                 ],
                 'stream' => [
                     'href' => $id,
-                    'supportedContentTypes' => array_keys($this->transformers),
+                    'supportedContentTypes' => self::SUPPORTED_CONTENT_TYPES,
```

Comment:
@prolic here I stopped, because LoadStream needs to know if a supported content type is given or not.
In case it is not given a description is returned (as a JsonResponse) that includes the supported content types.

Reply:
ok, so what's the problem?

Reply:
see my comment: https://github.com/prooph/event-store-http-middleware/pull/4#issuecomment-374267985

Reply:
also:
> But the problem is with inspecting the Accept headers. Which component should take over this task?

I really don't see the advantage of a response strategy bc it gives me nothing more than a response prototype. So why not just inject the response prototype in the first place?

I'm fine with the current solution:
We can inject transformers for different accept headers. So if we want to offer xml support later we just need to inject another transformer and send the correct accept headers.

Without transformers the request handler itself has to decide which response format should be used. I don't think it is the responsibility of the request handler to decide on the response type.

But this is only the happy path. Now what about the error cases like no supported accept header given? The thing is: we should remove all this checks because this should be done by a middleware in front of the request handler, but we cannot return an array or DTO from the request handler like it would be possible within an MVC framework.
I really don't see a good way of solving this problem and a response strategy just adds an additional piece of code to the problem without any benefit. At least I don't see the benefit.
But using a PSR proposal might be a good idea :thinking: